modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -527,7 +527,15 @@ func (suite *KeeperTestSuite) TestChanUpgradeAck() {
 		{
 			"fails due to upgrade incompatibility",
 			func() {
+				// Need to set counterparty upgrade in state and update clients to ensure
+				// proofs submitted reflect the altered upgrade.
 				counterpartyUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
+				path.EndpointB.SetChannelUpgrade(counterpartyUpgrade)
+
+				err := path.EndpointB.UpdateClient()
+				suite.Require().NoError(err)
+				err = path.EndpointA.UpdateClient()
+				suite.Require().NoError(err)
```

Comment:
We don't need to update chainA's client on chainB. Only EndpointA's lightclient needs to be updated. 

You can change this to use:

```suggestion
				suite.coordinator.CommitBlock(suite.chainB)

				err := path.EndpointA.UpdateClient()
				suite.Require().NoError(err)
```